src/Features/CSharp/Portable/Completion/CompletionProviders/ObjectCreationCompletionProvider.cs

```diff
@@ -145,5 +147,43 @@ protected override CompletionItemRules GetCompletionItemRules(IReadOnlyList<ISym
 
             return s_defaultRules;
         }
+
+        public override Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = null, CancellationToken cancellationToken = default)
+        {
+            var insertionText = SymbolCompletionItem.GetInsertionText(item);
+            if (commitKey == ';')
+            {
+                var endOfInsertionText = item.Span.Start + insertionText.Length;
+                var textChange = new TextChange(item.Span,insertionText + "();");
+                var completionChange = CompletionChange.Create(textChange,
+                    SymbolCompletionItem.GetPutCaretBetweenParenthesis(item)
+                     ? endOfInsertionText + 1
+                     : endOfInsertionText + 3, includesCommitCharacter: true);
+                return Task.FromResult(completionChange);
+            }
+
+            var insertionTextChange = new TextChange(item.Span, insertionText);
+            return Task.FromResult(CompletionChange.Create(insertionTextChange));
+        }
+
+        protected override CompletionItem CreateItem(
+            CompletionContext completionContext,
+            string displayText,
+            string displayTextSuffix,
+            string insertionText,
+            List<ISymbol> symbols,
+            SyntaxContext context,
+            bool preselect,
+            SupportedPlatformData supportedPlatformData)
+        {
+            var item = base.CreateItem(completionContext,
+               displayText, displayTextSuffix, insertionText, symbols,
+               context, preselect,
+               supportedPlatformData);
+
+            return SymbolCompletionItem.AddPutCaretBetweenParenthesis(
```

Comment:
I think we have kept all symbols (encoded) as part of the CompletionItem from this provider, by delay the check to GetChangeAsync, we can avoid the cost of calling `ShouldPutCaretBetweenParenthesis` on symbols of all items upfront (which might cause delay of showing completion list) to just one item when it's actually committed.